lib/PuppeteerSharp.Tests/WorkerTests/WorkerTests.cs

```diff
@@ -14,17 +15,22 @@ public WorkerTests(ITestOutputHelper output) : base(output)
         [Fact]
         public async Task PageWorkers()
         {
-            var pageCreatedCompletion = new TaskCompletionSource<bool>();
-            Page.WorkerCreated += (sender, e) => pageCreatedCompletion.TrySetResult(true);
+            var workerCreatedTcs = new TaskCompletionSource<bool>();
+            var workerDestroyedTcs = new TaskCompletionSource<bool>();
+
+            Page.WorkerCreated += (sender, e) => workerCreatedTcs.TrySetResult(true);
+            Page.WorkerDestroyed += (sender, e) => workerDestroyedTcs.TrySetResult(true);
+
             await Task.WhenAll(
-                    pageCreatedCompletion.Task,
-                    Page.GoToAsync(TestConstants.ServerUrl + "/worker/worker.html"));
+                workerCreatedTcs.Task,
+                Page.GoToAsync(TestConstants.ServerUrl + "/worker/worker.html"));
             var worker = Page.Workers[0];
             Assert.Contains("worker.js", worker.Url);
 
             Assert.Equal("worker function result", await worker.EvaluateExpressionAsync<string>("self.workerFunction()"));
 
             await Page.GoToAsync(TestConstants.EmptyPage);
+            await workerDestroyedTcs.Task.WithTimeout(1000);
```

Comment:
why the timeout? what will happen if we just `await`?

Reply:
If we fail because the worker is never destroyed I'd rather get an invalid worker on the next line than hanging the CI because the task never completes.

Reply:
Sorry, We wouldn't get to the next line because we would get a TimeoutException, but I'd rather get that than hanging the CI.